### PR TITLE
Reduce place queries

### PR DESCRIPTION
**Pull request by joelchoo**
Hey @LoneRifle @xkjyeah @shen-xujing can help me take a look at this and let me know what you think?

It's not a complete PR, just want to see if at this point anyone of you have strong opinions on how I'm doing this? I feel like there should be a better way to do this, but don't know enough about angular to be sure.

### CODE REVIEW

**Comment by xkjyeah** [comment]
Could you briefly explain the concept of what you are trying to do?

**Comment by joelchoo** [comment]
At a high level, we wanted to only do a place query when the normal text results are less than a certain length. 

this implementation involves not doing the place query in auto complete so that forces a text search first. Then if the text search results are too small, do the place query.

One tricky thing is how to handle this for multiple types of routes I.e. if say both the crowd start and the normal routes have Insufficient results we want to avoid doing two place queries.

**Comment by xkjyeah** [comment]
Oh... It looks like you might not be solving the problem. allRoutes is the list of all 100+ routes, so it will never be less than 3.

There's a bit of code duplication between crowdstart, lite and regular routes at the moment. Would you want to work on that first?

**Comment by joelchoo** [comment]
I use the allRoutes that is after the filter, so it is less than 3 if it doesn't match anything in the text search! I've tested this and pretty sure it works.

I will look at the code duplication too

**Comment by xkjyeah** [comment]
Oops sorry. I think I understand your code now.

Essentially you want to:
1. When query changes
2. Do a textual filter
3. If the textual filter fails do a places query

Currently you are shunting step 2 and 3 to a more distant part of the code.

You should handle all of that within the queryText watcher

**Comment by joelchoo** [comment]
Ok thanks I'll check that out!

**Comment by xkjyeah** [comment]
Otherwise... It sounds like a reasonable way to reduce the number of queries. Ok, pls proceed then

**Comment by joelchoo** [comment]
Okay this is ready for review. If approved, can merge and see if it helps with reducing the place queries.

@xkjyeah 

> Currently you are shunting step 2 and 3 to a more distant part of the code.
> You should handle all of that within the queryText watcher

Right now, step 2 (textual filter) is being done in a whole bunch of places, handled separately for each different type of route (e.g. activated crowdstart, unactivated crowdstart, normal, lite).

I agree that we should refactor that to be handled within the queryText watcher. However, this PR is just a small test to see if we can reduce the number of places queries, and I don't want to try and make too many changes at the same time.

I'll do the refactor in the separate PR after this! :)

File: beeline/controllers/RoutesListController.js [hunk 1]
```diff
@@ -63,32 +66,13 @@ export default function(
...
-        }
-        // Otherwise return the fully formed place
-        place = _.assign(place,result);
-        // Return the found place
-        $scope.data.placeQuery =  place;
-        $scope.data.isFiltering = false;
-        $scope.$digest();
-      });
+      $scope.data.getPlaceDetails = false;
+      $scope.data.prediction = predictions[0];
```
Comment (shen-xujing) [inline beeline/controllers/RoutesListController.js]: predictions could be null due to error or no result found

**Comment by xkjyeah** [comment]
I cannot agree with watching a Boolean value (getPlaceDetails).

Just for the regular routes part, could you write a watcher that does all
three steps? Structure the watchers they way you might structure computed
properties in Vue.

You don't have to rewrite the other watchers (lite, crowdstart).

**Comment by joelchoo** [comment]
See #415

**Pull request closed without merging**